geomesa-core/src/main/scala/org/locationtech/geomesa/core/index/STIdxStrategy.scala

```diff
@@ -119,7 +119,19 @@ class STIdxStrategy extends Strategy with Logging with IndexFilterHelpers {
     qp.copy(table = table, iterators = iterators, numThreads = numThreads, hasDuplicates = hasDupes)
   }
 
-  def getSTIIIterCfg(iteratorConfig: IteratorConfig,
+  def decomposeToGeometry(f: Filter): Seq[Geometry] = f match {
+    case bbox: BBOX =>
+      val bboxPoly = bbox.getExpression2.asInstanceOf[Literal].evaluate(null, classOf[Geometry])
+      Seq(bboxPoly)
+    case gf: BinarySpatialOperator =>
+      extractGeometry(gf)
+    case or: OrImpl =>
```

Comment:
I thought that, before we could get to this routine, the top-most query had been transformed into DNF, and ```getQueryPlan``` was called on each of the top-level children.  If that's so, how can there be an ```OrImpl``` here?